services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_user_services.py

```diff
@@ -0,0 +1,190 @@
+    # Starts dynamic SIDECAR -------------------------------------
+    # creates a docker compose spec given the service key and tag
+    # fetching project form DB and fetching user settings
+
+    director_v0_client: DirectorV0Client = get_director_v0_client(app)
+    simcore_service_labels: SimcoreServiceLabels = (
+        await director_v0_client.get_service_labels(
+            service=ServiceKeyVersion(
+                key=scheduler_data.key, version=scheduler_data.version
+            )
+        )
+    )
+
+    groups_extra_properties = get_repository(app, GroupsExtraPropertiesRepository)
+    assert scheduler_data.product_name is not None  # nosec
+    allow_internet_access: bool = await groups_extra_properties.has_internet_access(
+        user_id=scheduler_data.user_id, product_name=scheduler_data.product_name
+    )
+
+    compose_spec: str = assemble_spec(
+        app=app,
+        service_key=scheduler_data.key,
+        service_version=scheduler_data.version,
+        paths_mapping=scheduler_data.paths_mapping,
+        compose_spec=scheduler_data.compose_spec,
+        container_http_entry=scheduler_data.container_http_entry,
+        dynamic_sidecar_network_name=scheduler_data.dynamic_sidecar_network_name,
+        swarm_network_name=scheduler_data.dynamic_sidecar.swarm_network_name,
+        service_resources=scheduler_data.service_resources,
+        has_quota_support=dynamic_sidecar_settings.DYNAMIC_SIDECAR_ENABLE_VOLUME_LIMITS,
+        simcore_service_labels=simcore_service_labels,
+        allow_internet_access=allow_internet_access,
+        product_name=scheduler_data.product_name,
+        user_id=scheduler_data.user_id,
+        project_id=scheduler_data.project_id,
+        node_id=scheduler_data.node_uuid,
+        simcore_user_agent=scheduler_data.request_simcore_user_agent,
+        swarm_stack_name=dynamic_sidecar_settings.SWARM_STACK_NAME,
+    )
+
+    _logger.debug(
+        "Starting containers %s with compose-specs:\n%s",
+        scheduler_data.service_name,
+        compose_spec,
+    )
+
+    async def progress_create_containers(
+        message: str, percent: PositiveFloat, task_id: TaskId
+    ) -> None:
+        # TODO: detect when images are pulling and change the status
+        # of the service to pulling
+        _logger.debug("%s: %.2f %s", task_id, percent, message)
+
+    # data from project
+    projects_repository = get_repository(app, ProjectsRepository)
+    project: ProjectAtDB = await projects_repository.get_project(
+        project_id=scheduler_data.project_id
+    )
+    project_name = project.name
+    node_name = project.workbench[NodeIDStr(scheduler_data.node_uuid)].label
+
+    # data from user
+    users_repository = get_repository(app, UsersRepository)
+    user_email = await users_repository.get_user_email(scheduler_data.user_id)
+
+    metrics_params = CreateServiceMetricsAdditionalParams(
+        wallet_id=UNDEFINED_INT_METADATA,
+        wallet_name=UNDEFINED_STR_METADATA,
```

Comment:
Aa now I get it, this just an temporary solution, until we will have the wallets ready?